src/wui/wui_api.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * wui_api.c
+ * \brief
+ *
+ *  Created on: Jan 24, 2020
+ *      Author: joshy <joshymjose[at]gmail.com>
+ */
+
+#include "wui_api.h"
+
+#include "wui.h"
+#include "filament.h"
+
+#include "cmsis_os.h"
+#include "stdarg.h"
+
+#define BDY_WUI_API_BUFFER_SIZE 512
+#define BDY_NO_FS_FLAGS    0   // no flags for fs_open
+#define BDY_API_PRINTER_LEN     12 // length of "/api/printer" string
+#define BDY_API_JOB_LEN     8   // length of "/api/job" string
+#define X_AXIS_POS 0
+#define Y_AXIS_POS 1
+#define Z_AXIS_POS 2
+// for data exchange between wui thread and HTTP thread
+extern marlin_vars_t webserver_marlin_vars;
+extern osMutexId wui_web_mutex_id;
+static marlin_vars_t webserver_marlin_vars_copy;
+// for storing /api/* data
+static struct fs_file api_file;
+static char _buffer[BDY_WUI_API_BUFFER_SIZE];
+
+static int char_streamer(const char *format, ...) {
+    int rv = 0;
+    va_list args;
+    va_start(args, format);
+    rv = vsnprintf(_buffer, BDY_WUI_API_BUFFER_SIZE, format, args);
+    va_end(args);
+    return rv;
+}
+
+static void wui_api_job(struct fs_file* file) {
+
+    const char* file_name = "test.gcode";
+    uint8_t sd_percent_done = (uint8_t)(webserver_marlin_vars_copy.sd_percent_done);
+    uint32_t print_duration = (uint32_t)(webserver_marlin_vars_copy.print_duration);
+
+    int response_len = char_streamer("{"
+            "\"file\":\"%s\","
+            "\"total_print_time\":%d, "
+            "\"progress\":{\"precent_done\":%d}"
+            "}",
+            file_name,
+            print_duration,
+            sd_percent_done
+            );
+    file->len = response_len;
+    file->data = (const char*)&_buffer;
+    file->index = response_len;
+    file->pextension = NULL;
+    file->flags = BDY_NO_FS_FLAGS;
+}
+
+static void wui_api_printer(struct fs_file* file) {
+
+    int32_t actual_nozzle = (int32_t) (webserver_marlin_vars_copy.temp_nozzle);
```

Comment:
/nitpick/ "current temperature", not "actual temperature"

Reply:
this has something to do with the then definition of "api" definitions. not my invention

Reply:
actual temperature is not proper English
if that's in an API of ours, the API must be changed

Reply:
this will be handled in future